### ISSUE TITLE
Warning: React.findDOMNode is deprecated. Please use ReactDOM.findDOMNode from require('react-dom') instead.

### ISSUE THREAD

**Issue by pzavolinsky** [effektif/react-mentions#58]
Hi, I'm getting this warning here:
```js
  updateSuggestionsPosition: function updateSuggestionsPosition() {
    if (!this.refs.caret || !this.refs.suggestions) return;

    var containerEl = this.refs.container;
    var caretEl = this.refs.caret;
    var suggestionsEl = React.findDOMNode(this.refs.suggestions); // <<<<<< Warning here.
    var highligherEl = this.refs.highlighter;
    if (!suggestionsEl) return;

    var leftPos = caretEl.offsetLeft - highligherEl.scrollLeft;
    // guard for mentions suggestions list clipped by right edge of window
    if (leftPos + suggestionsEl.offsetWidth > containerEl.offsetWidth) {
      suggestionsEl.style.right = "0px";
    } else {
      suggestionsEl.style.left = leftPos + "px";
    }
    suggestionsEl.style.top = caretEl.offsetTop - highligherEl.scrollTop + "px";
  },
```

Can potentially be fixed by using:
```js
    var suggestionsEl = ReactDOM.findDOMNode(this.refs.suggestions);
```

**Comment by coryvirok** [comment]
Ack, this is probably my fault with https://github.com/effektif/react-mentions/pull/56 Should probably just list react-dom as a dependency instead of my PR.

**Comment by frontendphil** [comment]
My guess would be that we can find a way to update the style without figuring out the DOM node first. I would really like to leave react-dom out as a dependency because it ads a whopping 500kb to the build.

**Comment by frontendphil** [comment]
So ignore the warning for now, we leave this issue open and try to find a solution in the meantime.

**Comment by evankline** [comment]
@frontendphil any update on this issue? Does it make sense to add React-DOM as a dependency until a different solution is found?

**Comment by frontendphil** [comment]
As we need to support NPM 3 at some time, we probably need to depend on `react` and `react-dom` explicitly anyways. So probably we need to add those as a dependency. Only problem I see is with the combined build. There I really don't want to include the react and react-dom sources. I'm open for solutions on how to exclude them there.

**Comment by jfschwarz** [comment]
I would vote for adding NPM 3 support, dropping Bower support, and no longer providing a combined build with version 0.3.

**Comment by pzavolinsky** [comment]
Hi @frontendphil, if you are using the combined build (that is my case btw) is probably safe to assume that you are including those dependencies on your own (and leaking them into the global scope like window.ReactDOM or global.ReactDOM).

Maybe you could just check for those globals.

**Comment by evankline** [comment]
@pzavolinsky is most likely correct, otherwise I'd agree with @jfschwarz and drop Bower support as NPM 3 is probably a more adequate solution moving forward anyway.

**Comment by evankline** [comment]
Any updates on this issue being resolved?

**Comment by fremn** [comment]
@pzavolinsky leaking into global/window space doesnt seem kosher  

+1 @evankline question

**Comment by pzavolinsky** [comment]
@fremn :) true, although a combined build is all about leaking stuff into global/window.

In any case, I'm not suggesting leaking stuff but rather taking advantage of the fact that:
  - the only sensible usage of `react-mentions`'s combined build is after `react-dom`'s combined build.
  - the combined build for `react-dom` already leaks the missing dependency.

That is:

```html
<script src="https://fb.me/react-dom-0.14.6.js"></script>
<script>
  // ReactDOM was leaked into window:
  ReactDOM.findDOMNode !== undefined
</script>
<script src="react-mentions-combined-build.js"></script>
```

The full-blown hack could be more or less:

```js
var reactDOM = typeof global !== "undefined" && global.ReactDOM
  || typeof window !== "undefined" && window.ReactDOM
  || require('react-dom');
```

That being said, I do agree that this type of solution is somewhat post-apocalyptic and an alternative that removes the need for `ReactDOM.findDOMNode` would make the world a better place.

**Comment by styks1987** [comment]
Running into this too.

**Comment by johannesnagl** [comment]
> var suggestionsEl = React.findDOMNode(this.refs.suggestions);

doenst `this.refs.suggestions` nowadays (react 0.14+) returns the dom element on it's own? According to the docs: https://facebook.github.io/react/docs/top-level-api.html#reactdom.finddomnode

> In most cases, you can attach a ref to the DOM node and avoid using findDOMNode at all. 

So can't we simply use `var suggestionsEl = this.refs.suggestions;`?

**Comment by sisardor** [comment]
Any solution for this?

**Comment by jfschwarz** [comment]
Fixed in v0.3.0
